crutest/src/main.rs

```diff
@@ -3993,41 +4048,100 @@ async fn demo_workload(
 
 /*
  * This is a test workload that generates a single write spanning an extent
- * then will try to read the same.
+ * then will try to read the same.  When multiple sub_volumes are present we
+ * also issue a Write/Flush/Read that will span the two sub_volumes.
  */
 async fn span_workload(volume: &Volume, ri: &mut RegionInfo) -> Result<()> {
-    /*
-     * Pick the last block in the first extent
-     */
-    let block_index = (ri.extent_size.value - 1) as usize;
+    let mut extent_block_start = 0;
+    let last_sub_volume = ri.volume_info.volumes.len() - 1;
+    let block_size = ri.volume_info.block_size;
+    for (index, sv) in ri.volume_info.volumes.iter().enumerate() {
+        // Pick the last block in the first extent
+        let extent_size = sv.blocks_per_extent as usize;
+        let block_index = extent_block_start + extent_size - 1;
+
+        // Update the counter for the blocks we are about to write.
+        ri.write_log.update_wc(block_index);
+        ri.write_log.update_wc(block_index + 1);
```

Comment:
Why does this not fail on the last subvolume?  I'd expect it to write out of bounds.